src/steamship_langchain/vectorstores/steamship_vector_store.py

```diff
@@ -0,0 +1,106 @@
+import uuid
+from itertools import zip_longest
+from typing import Any, Iterable, List, Optional
+
+from langchain.docstore.document import Document
+from langchain.vectorstores import VectorStore
+from steamship import Steamship, Tag
+
+
+class SteamshipVectorStore(VectorStore):
+    """Wrapper around Steamships vector database.
+
+    Example:
+        .. code-block:: python
+
+            from steamship_langchain import SteamshipVectorStore
+            faiss = SteamshipVectorStore(embedding_function, index, docstore)
+
+    """
+
+    def __init__(
+        self,
+        client: Steamship,
+        embedding: str,
+        index_name: str,
+    ):
+        """Initialize with necessary components."""
+
+        self.client = client
+        self.index_name = index_name or uuid.uuid4().hex
+
+        self.index = client.use_plugin(
+            plugin_handle="embedding-index",
+            instance_handle=self.index_name,
+            config={
+                "embedder": {
+                    "plugin_handle": "openai-embedder",
+                    "instance_handle": self.index_name,
+                    "fetch_if_exists": True,
+                    "config": {"model": embedding, "dimensionality": 1536},
```

Comment:
question: does dimensionality need to be a param ?  is it tied to the `embedding` ?

Reply:
after offline convo, my understanding is that this will now be a param to the `__init__`. will approve to unblock merge with that change in the timezone change-over ;)